examples/seq2seq/input_pipeline.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2022 The Flax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Input pipeline for seq2seq addition example."""
+
+import numpy as np
+import random
+
+import jax
+import jax.numpy as jnp
+
+
+class CharacterTable(object):
+  """Encode/decodes between strings and integer representations."""
+
+  def __init__(self, chars, max_len_query_digit = 3):
+    self._chars = sorted(set(chars))
+    self._char_indices = dict(
+        (ch, idx + 2) for idx, ch in enumerate(self._chars))
+    self._indices_char = dict(
+        (idx + 2, ch) for idx, ch in enumerate(self._chars))
+    self._indices_char[self.pad_id] = '_'
+    # Maximum length of a single input digit
+    self._max_len_query_digit = max_len_query_digit
+
+  @property
+  def pad_id(self):
+    return 0
+
+  @property
+  def eos_id(self):
+    return 1
+
+  @property
+  def vocab_size(self):
+    return len(self._chars) + 2
+
+  @property
+  def max_input_len(self):
+    """Returns the max length of an input sequence."""
+    return self._max_len_query_digit * 2 + 2  # includes EOS
+
+  @property
+  def max_output_len(self):
+    """Returns the max length of an output sequence."""
+    return self._max_len_query_digit + 3  # includes start token '=' and EOS.
+
+  @property
+  def encoder_input_shape(self):
+    max_input_len = self._max_len_query_digit * 2 + 2  # includes EOS
+    return (1, max_input_len, self.vocab_size)
+
+  @property
+  def decoder_input_shape(self):
+    return (1, self.max_output_len, self.vocab_size)
+
+  def encode(self, inputs):
+    """Encode from string to list of integers."""
```

Comment:
nit: function pydoc is "descriptive style" i.e. `Encodes` and not `Encode` (here and elsewhere)